source/helpers/conventional-commits.ts

```diff
@@ -1,26 +1,50 @@
-import zipTextNodes from 'zip-text-nodes';
+import {assertNodeContent} from './dom-utils.js';
 
 // Using https://www.conventionalcommits.org/ as a reference.
-const CONVENTIONAL_COMMIT_REGEX = /^(?<type>\w+)(?:\((?<scope>.+)\))?: /;
-
-export function parseConventionalCommit(commitTitle: string): RegExpExecArray | undefined {
-	return CONVENTIONAL_COMMIT_REGEX.exec(commitTitle) ?? undefined;
+const conventionalCommitRegex = /^(?<type>\w+)(?:\((?<scope>.+)\))?: /;
+
+const types = new Map([
+	['feat', 'Feature'],
+	['fix', 'Fix'],
+	['chore', 'Chore'],
+	['docs', 'Docs'],
+	['build', 'Build'],
+	['refactor', 'Refactor'],
+	['test', 'Test'],
+	['ci', 'CI'],
+	['perf', 'Performance'],
+]);
+
+export function parseConventionalCommit(commitTitle: string): {
+	rawType: string;
+	type: string;
+	scope?: string;
+	raw: string;
+} | undefined {
+	const match = conventionalCommitRegex.exec(commitTitle);
+	if (!match?.groups?.type) {
+		return;
+	}
+
+	const {type: rawType, scope} = match.groups;
+	const type = types.get(rawType);
+	if (!type) {
+		return;
+	}
+
+	return {
+		rawType,
+		type,
+		scope: scope ? `${scope}: ` : undefined,
+		raw: match[0],
+	};
 }
 
-export function removeCommitAndScope(
-	element: HTMLElement,
-	match: RegExpExecArray,
-): void {
-	const {type, scope} = match?.groups ?? {};
-
-	const semanticCommitTypeAndScope = scope ? `${type}(${scope}):` : `${type}:`;
-
-	const modified = element.textContent.replace(semanticCommitTypeAndScope, match => `<s>${match}</s>`);
-
-	const temporaryDiv = document.createElement('div');
-	temporaryDiv.innerHTML = modified;
-
-	zipTextNodes(element, temporaryDiv);
-
-	element.querySelector('s')!.remove();
+/**
+Remove the raw commit prefix from the single text node.
+Note: It does not support pre-formatted titles like: fix(#1 `x`)
+*/
+export function removeCommitAndScope(textNode: Text | ChildNode): void {
+	assertNodeContent(textNode, conventionalCommitRegex);
+	textNode.textContent = textNode.textContent.replace(conventionalCommitRegex, '');
```

Comment:
<table>
<tr>
	<th>Before
	<th>After
<tr>
	<td><img width="298" alt="Screenshot 1" src="https://github.com/user-attachments/assets/438e3331-5177-4b30-9627-fd97dd748d73">
	<td><img width="291" alt="Screenshot 6" src="https://github.com/user-attachments/assets/8df7aae1-d899-4d26-8336-4e239c02abea">
</table>

Reply:
This indeed breaks support for references and backticks, but also restores the links on those issue references (which is preferable anyway)


<table>
<tr>
	<th>Before
	<th>After
<tr>
	<td><img width="269" alt="Screenshot 5" src="https://github.com/user-attachments/assets/78080224-1845-450d-8e9a-9e2ebaaae675">
	<td><img width="269" alt="Screenshot 7" src="https://github.com/user-attachments/assets/3d6e8122-a0db-4d32-ab81-9cac1c0aabb5">


</table>

Reply:
I don't know how I feel about the after. Maybe you missed this https://github.com/refined-github/refined-github/pull/7796#issuecomment-2357085761.

Additionally you add arbitrary cursive formatting to the scope.

 > This indeed breaks support for references and backticks, but also restores the links on those issue references (which is preferable anyway)

Is ther no way to retain references and backticks in the label? Pretty sure you can just wrap all the elements as a child of the label element.

Reply:
I did not miss those, I just disagreed. 

> arbitrary cursive formatting to the scope.

If you look at a conventional changelog, you’ll see that the scope is bold. Since the whole commit title is bold, I made it cursive. Hardly arbitrary. 

> Is ther no way to retain references and backticks in the label? Pretty sure you can just wrap all the elements as a child of the label element.

There sure is, but it’s hard and it already took days of reviews to pass a barebones version, I don’t want to spend days reviewing code that aims to preserve the DOM without drawbacks.

Reply:
> Since the whole commit title is bold, I made it cursive

This is arbitrary. Why not make it normal? Why not make it uppercase? Why cursive?

> There sure is, but it’s hard and it already took days of reviews to pass a barebones version, I don’t want to spend days reviewing code that aims to preserve the DOM without drawbacks.

The only difference that needs to be done is to collect all the elements to an array until the colon , and then add as content of the label.

Reply:
In typography bold and italics are used to give emphasis. Again, since the whole title is already bold, I used the second style. 

> The only difference that needs to be done is to collect all the elements to an array until the colon , and then add as content of the label.

Easier said than done. I’m done discussing this.

Reply:
Unfortunately, I am not comfortable with this change due to the reasons I have mentioned, so I'll stop the discussion and move forward with a fork that reflects the changes I have in mind. Thanks regardless.